test/framework/src/main/java/org/elasticsearch/cluster/metadata/DataStreamTestHelper.java

```diff
@@ -494,7 +494,7 @@ public static MetadataRolloverService getMetadataRolloverService(
         AllocationService allocationService = mock(AllocationService.class);
         when(allocationService.reroute(any(ClusterState.class), any(String.class), any())).then(i -> i.getArguments()[0]);
         when(allocationService.getShardRoutingRoleStrategy()).thenReturn(TestShardRoutingRoleStrategies.DEFAULT_ROLE_ONLY);
-        MappingLookup mappingLookup = null;
+        MappingLookup mappingLookup = MappingLookup.EMPTY;
```

Comment:
Found another bug in the test code. This `mappingLookup` instance is used to setup `DocumentMapper` & `MapperService` mocks via a `mockIndicesServices` call (see line 518 of this file).

When `dataStream` is `null`, this method sets `mappingLookup` to `null`, which would never happen in production. See the relevant production code:

- [DocumentMapper](https://github.com/elastic/elasticsearch/blob/8a1db8c6c3ffa8028b98c9c1393aee9d9ea4d6f5/server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java#L41)
- [MapperService](https://github.com/elastic/elasticsearch/blob/68093e6e0dd2d69bf250227646df3eb077f01c2a/server/src/main/java/org/elasticsearch/index/mapper/MapperService.java#L637)

Changing the default `mappingLookup` value to `MappingLookup.EMPTY` accurately reflects the behavior of production code and fixes the NPEs.